lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -28,10 +28,20 @@
 class PDOStatement extends \PDOStatement implements Statement
 {
     /**
+     * @var PDOConnection|null
+     */
+    private $connection;
+
+    /**
      * Protected constructor.
+     *
+     * @param PDOConnection|null $connection
+     *
+     * @todo do we need to keep BC here? Or can we make $connection mandatory?
```

Comment:
We have to maintain BC in my opinion, otherwise we may break existing manual DIC calls.